doc/hardware/porting/board_porting.rst

```diff
@@ -354,7 +354,10 @@ named ``plank``:
 
 :file:`plank_defconfig`
   A Kconfig fragment that is merged as-is into the final build directory
-  :file:`.config` whenever an application is compiled for your board.
+  :file:`.config` whenever an application is compiled for your board. The name
+  of this file also determines the name of the board as seen by the build system
+  in the case where the ``BOARD`` symbol was not provided by the user (such as with
+  the build option ``-DBOARD=<board>``).
```

Comment:
not sure that helped.

Users must give a board when invoking a build, else you get an error like this:
```
CMake Error at /projects/github/ncs/zephyr/cmake/modules/extensions.cmake:2501 (message):
  BOARD is not being defined on the CMake command-line, in the environment or
  by the app.
```

Reply:
Do you have any suggestion about where/what the clarification about the board name should be in the docs?

Reply:
I gave a recommendation here:
https://github.com/zephyrproject-rtos/zephyr/pull/51011#discussion_r991999587

> But this should be fixed by improving the overall description of the board folder vs. the actual board, not by making imprecise description of the files describing the actual board.

So probably at the start of this:
https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#create-your-board-directory

Either update the existing text, or create a note.

In case of a note, could be something along:
```
.. note::

   A board and the board dir can be named differently. In fact a board dir can contain multi boards,
   this is used for multi-core SoC where the board folder may be named `plank`, but each core is
   referenced according to the SoC, like:
   `boards/<ARCH>/plank` may hold files for `plank_core_a`, `plank_core_b`, etc.

   Each logical board in such a setup must contain all individual files described below for each core.
```

Feel free to adjust the description in any way you like, which you believe would have made it easier to understand the behavior.

Reply:
sorry about that, I misunderstood what you meant before but I reread it and now I get your points

I updated the PR, do you agree with the new changes?